pkg/agent/imagebuilder/content.go

```diff
@@ -37,11 +37,8 @@ type ConfigBuilder struct {
 }
 
 // New creates a new ConfigBuilder by reading the ZTP manifests
-func New() (*ConfigBuilder, error) {
-	pullSecret, err := manifests.GetPullSecret()
-	if err != nil {
-		return nil, err
-	}
+func New(agentManifests agentAssets.AgentManifests) (*ConfigBuilder, error) {
```

Comment:
My personal preference would be to avoid pushing too deep the assets in the code, as this will increase the coupling between them and the agent code (agentManifests could be easily replaced by a locator interface for that in case), but passing directly the required concrete types (and probably files). I agree it could be fine for now, and we'll see how it will evolve